src/GapicClientTrait.php

```diff
@@ -785,6 +797,19 @@ private function configureCallConstructionOptions(string $methodName, array $opt
         ];
     }
 
+    /**
+     * @return array
+     */
+    private function configureCallOptions(array $optionalArgs): array
+    {
+        if ($this->isNewClientSurface()) {
```

Comment:
Is this worth caching this result in a property on init if it will be invoked on every API call?

Reply:
for sure

Reply:
done